### DISCUSSION TITLE
redact document script: set_metadata() is not working

### DISCUSSION THREAD

**Discussion by rockerjunge** [pymupdf/PyMuPDF#3236] [Q&A]
### Description of the bug

I'm new to programming and still finding my way around, but I've encountered an issue with the set_metadata() function. Unfortunately, I can't seem to get it to work correctly. Below is my code where I've implemented this method. Despite my efforts, both print statements yield identical results, with the metadata still present.

### How to reproduce the bug

    import os
    import fitz
    def test_method(input_dir, output_dir):
        for filename in os.listdir(input_dir):
            if filename.endswith(".pdf"):
                pdf_path = os.path.join(input_dir, filename)
                output_path = os.path.join(output_dir, filename.replace(".pdf", "_test.pdf"))
    
                doc = fitz.open(pdf_path)
               
                print(doc.metadata)
                doc.set_metadata({})
                print(doc.metadata)
    
                doc.save(output_path, garbage=4)
                doc.close()
    
    input_dir = "H:\\Schwärzer Test"
    output_dir = "H:\\Schwärzer Test\\Geschwaerzt"
    
    if not os.path.exists(output_dir):
        os.makedirs(output_dir)
    
    test_method(input_dir, output_dir)

### PyMuPDF version

1.23.26

### Operating system

Windows

### Python version

3.11

**Comment by JorjMcKie** [comment]
Oops - I am sorry:
You are quite right - this is a bug!

**Comment by JorjMcKie** [comment]
The problem is not too bad however:
The change has landed in the document, which you can confirm by saving and reopening.
Only the `.metadata` dictionary is not properly updated.

**Comment by rockerjunge** [comment]
Oh okay. So I´d have a second question. I imported fitz, as you can see. And i installed the PyMuPDF module on PyCharm. Still, with the code
"doc.set_metadata(new_metadata)" 
PyCharm gives me the error code: 
"Unresolved attribute reference 'set_metadata' for class 'Document'"

Same is with Scrubs, PyCharm can´t find any reference.

**Reply by JorjMcKie** [reply]
We do not chase the different complaints that PyCharm and friends may have.
All the IDE behave differently and many have errors - including especially PyCharm!
So that is something that will have to wait ...

**Comment by julian-smith-artifex-com** [comment]
The related issue #3237 has been fixed by #3555.
